Softeq.XToolkit.PushNotifications.iOS/IosNotificationCategoriesProvider.cs

```diff
@@ -0,0 +1,135 @@
+        /// <param name="options">Additional options for handling notifications of this type</param>
+        protected void AddCategory(string categoryId, string[] intentIdentifiers, string hiddenPreviewsBodyPlaceholder,
+            UNNotificationCategoryOptions options)
+        {
+            _actionsForCategories.TryGetValue(categoryId, out var actions);
+
+            var messageCategory = UNNotificationCategory.FromIdentifier(categoryId,
+                actions?.ToArray() ?? new UNNotificationAction[] { }, intentIdentifiers, hiddenPreviewsBodyPlaceholder, options);
+            NotificationCategories.Add(messageCategory);
+        }
+
+        /// <summary>
+        /// A method to create and save a Category that will be registered for your app with all the actions
+        /// that were previously added for this Category Id and with hiddenPreviewsBodyPlaceholder and categorySummaryFormat.
+        /// NOTE: CategorySummaryFormat is only supported since iOS 12. For iOS 11 category will be created ignoring this option.
+        /// See <see cref="UNNotificationCategory.FromIdentifier"/>
+        /// </summary>
+        /// <param name="categoryId">The unique identifier for the Category. Should not be empty</param>
+        /// <param name="intentIdentifiers">The intent identifier strings that you want to associate with notifications of this type.
+        /// The Intents framework defines constants for each type of intent that you can associate with your notifications</param>
+        /// <param name="hiddenPreviewsBodyPlaceholder">A placeholder string to display when the user has disabled notification previews for the app.
+        /// Include the characters %u in the string to represent the number of notifications with the same thread identifier</param>
+        /// /// <param name="categorySummaryFormat">Category’s summary format string.
+        /// Include the characters %u in the string to represent the number of notifications with the same thread identifier</param>
+        /// <param name="options">Additional options for handling notifications of this type</param>
+        protected void AddCategory(string categoryId, string[] intentIdentifiers, string hiddenPreviewsBodyPlaceholder,
+            string categorySummaryFormat, UNNotificationCategoryOptions options)
+        {
+            _actionsForCategories.TryGetValue(categoryId, out var actions);
+
+            UNNotificationCategory messageCategory;
+
+            if (UIDevice.CurrentDevice.CheckSystemVersion(12, 0))
+            {
+                messageCategory = UNNotificationCategory.FromIdentifier(categoryId,
+                    actions?.ToArray() ?? new UNNotificationAction[] { }, intentIdentifiers,
+                    hiddenPreviewsBodyPlaceholder, new NSString(categorySummaryFormat), options);
+            }
+            else
+            {
+                messageCategory = UNNotificationCategory.FromIdentifier(categoryId,
```

Comment:
Seems like in this case logic is completely duplicates already existed AddCategory() method (the one without categorySummaryFormat). Can we check system version first and if iOS version is less than 12.0 - just cal that method?

Reply:
Done